include/openthread/udp.h

```diff
@@ -115,7 +115,8 @@ typedef void (*otUdpReceive)(void *aContext, otMessage *aMessage, const otMessag
 typedef enum otNetifIdentifier
 {
     OT_NETIF_UNSPECIFIED = 0, ///< Unspecified network interface.
-    OT_NETIF_THREAD,          ///< The Thread interface.
+    OT_NETIF_THREAD,          ///< The Thread interface - allow use of platform UDP.
+    OT_NETIF_THREAD_NATIVE,   ///< The native Thread interface (within OpenThread) - do not use platform UDP.
```

Comment:
How about something like:
- `OT_NETIF_THREAD_HOST`
- `OT_NETIF_THREAD_INTERNAL`

Reply:
👍 Sounds good to me. Changed in new push.